go/vt/vttablet/grpctmclient/client.go

```diff
@@ -648,17 +662,28 @@ func (client *Client) InitReplica(ctx context.Context, tablet *topodatapb.Tablet
 }
 
 // DemoteMaster is part of the tmclient.TabletManagerClient interface.
-func (client *Client) DemoteMaster(ctx context.Context, tablet *topodatapb.Tablet) (string, error) {
+func (client *Client) DemoteMaster(ctx context.Context, tablet *topodatapb.Tablet) (*replicationdatapb.MasterStatus, error) {
 	cc, c, err := client.dial(tablet)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	defer cc.Close()
 	response, err := c.DemoteMaster(ctx, &tabletmanagerdatapb.DemoteMasterRequest{})
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	return response.Position, nil
+	masterStatus := response.MasterStatus
+	if masterStatus == nil {
+		// We are assuming this means a response came from an older server.
+		masterStatus = &replicationdatapb.MasterStatus{
+			Position:             response.DeprecatedPosition,
+			FilePosition:         "",
+			XXX_NoUnkeyedLiteral: response.XXX_NoUnkeyedLiteral,
+			XXX_unrecognized:     response.XXX_unrecognized,
+			XXX_sizecache:        response.XXX_sizecache,
```

Comment:
I don't think we should copy these fields. As I understand it, these fields contain information to help preserve unknown child fields of the top-level `DemoteMasterResponse` message. That info is not relevant to this `MasterStatus` message since it's from a different type of message altogether.

We know we're not losing any unknown fields from `response.MasterStatus` because the whole thing was `nil` in this case.

Reply:
Ok, that's a good point. Will adjust now.

Reply:
Done. PTAL when you get a chance.